python/ray/autoscaler/_private/kubernetes/node_provider.py

```diff
@@ -71,7 +72,21 @@ def get_node_id(self, ip_address, use_internal_ip=True) -> str:
             raise ValueError("Must use internal IPs with Kubernetes.")
         return super().get_node_id(ip_address, use_internal_ip=use_internal_ip)
 
-    def set_node_tags(self, node_id, tags):
+    def set_node_tags(self, node_ids, tags):
+        while 1:
```

Comment:
I think it's better to only try a limited number of times (like 3)

Reply:
Good idea - just changed it to do three retries.